tests/main/snap-run-hook/task.yaml

```diff
@@ -0,0 +1,88 @@
+summary: Check that `snap run` can actually run hooks
+
+prepare: |
+    echo "Ensure we have a core snap with snap run"
+    $SPREAD_PATH/tests/lib/reset.sh
+    snap install --channel=candidate ubuntu-core
+    snapbuild $TESTSLIB/snaps/basic-hooks .
+    snap install basic-hooks_1.0_all.snap
+
+    # We want to use the in-tree snap-exec, not the one in the core snap. To
+    # accomplish that, we'll just unpack the core we just grabbed, shove the new
+    # snap-exec in there, and repack it.
+
+    # First of all, unmount the core
+    systemctl stop snapd.service
+    core="$(realpath /snap/ubuntu-core/current)"
+    snap="$(mount | grep "$core" | awk '{print $1}')"
+    umount "$core"
+
+    # Now unpack the core, inject the new snap-exec into it, and repack it.
+    unsquashfs "$snap"
+    cp /usr/lib/snapd/snap-exec squashfs-root/usr/lib/snapd/
+    mv "$snap" "${snap}.orig"
+    mksquashfs squashfs-root "$snap" -comp xz
```

Comment:
I would slightly prefer that we use `snapbuild` here. Its build as part of the tests and will build the squashfs with the right options. I don't think it makes a practical difference currently but it may make it easier in the future.